### DISCUSSION TITLE
Implementation in Tensorflow 1.15

### DISCUSSION THREAD

**Discussion by arxidinakbar** [danielegrattarola/spektral#178] [Q&A]
Hi,

I'm planning to use GAT layers on my own dataset, where implement in Tensorflow1.15's Graph without Keras API due to project needed.
The Input nodes are given in shape [2, N, 128], corresponding Adjacency matrices are [2, N, N].
I wanna know how should I use GraphAttention's batch mode without Keras 'Input'.
So, is it correct as below directly feeding the data.
```
node_buffer = [2, 2048, 128]
adj_buffer = [2, 2048, 2048]
droup1 = Dropout(node_buffer, rate=0.6, name='droup1')
gat_1 = GraphAttention(channels=64, attn_heads=4, dropout_rate=0.6, concat_heads=True, name='gat_1')([droup1, adj_buffer])
droup2 = Dropout(GAT_conv1, rate=0.6, name='droup2')
gat_2 = GraphAttention(channels=64, attn_heads=4, dropout_rate=0.6, concat_heads=True, name='gat_2')([droup2, adj_buffer])
```

I'm aware of old version don't maintain any more, so I'm wondering is it still functional?

Best luck,

**Comment by danielegrattarola** [comment]
If you don't want to use Input layers I think you need to enable eager execution manually in TF 1.15.
Then, I think it should work as you expect. Batch mode was supported by GAT in the TF1 version of Spektral.

Are you having any specific problem with the model?

Cheers
